docs/proposals/unified-backend-architecture/tech-spec.md

```diff
@@ -0,0 +1,239 @@
+# Technical Specification for Unified Backend Architecture
+
+## Author and Document Owner
+
+[Denis Shevchenko](https://iohk.io/team/denis-shevchenko/)
+
+## Status
+
+DRAFT
+
+## User Story
+
+[CHW-35](https://iohk.myjetbrains.com/youtrack/issue/CHW-35)
+
+## Motivation
+
+Currently Daedalus can be treated as a heavyweight client: it works in a pair with Cardano node which stores
+full copy of the blockchain. It leads three problems:
+
+1. Such a heavyweight client must have enough disk space to store growing blockchain.
+2. Cardano node requires quite a lot system resources.
+3. During (first) start heavyweight client must sync (new) blocks and this is long process, even with good
+   internet connection.
+
+These problems make impossible to install such a heavyweight client on a mobile device. The purpose of Unified
+Backend Architecture is to solve these problems.
+
+We propose to create a lightweight Daedalus client. And it is possible to unify the different use cases under a
+single architecture, starting from the crucial observation that both interfacing with an hardware wallet or
+provide a mobile client can be seen as a specialisation of the general case of having a "keyless" wallet backend
+operating with the private keys and cryptography "hosted" on the client side. It will allow to keep blockchain
+on a backend only.
+
+## Prerequisites
+
+The reader should be familiar with following concepts:
+
+1. Daedalus frontend,
+2. wallet backend web API,
+3. work with wallets,
+4. transaction processing.
+
+Also it is assumed that the reader knows basic terminology from [Cardano Docs](https://cardanodocs.com/),
+for example, UTXO.
+
+## Assumptions
+
+N/A, see **Prerequisites**.
+
+## Requirements
+
+Unification of the wallet backend architecture should allow to:
+
+1. work with different hardware wallets in the same manner,
+2. create lightweight Daedalus mobile clients (iOS/Android),
+3. create lightweight Daedalus desktop clients (Windows/macOS/Linux).
+
+The main benefit of such a lightweight client is independence from the growing blockchain: the client does not
+store blockchain locally and does not wait for a "Syncing blocks".
```

Comment:
Also ideally, the unified arch should also cover the existing desktop use case, running the wallet backend on the same machine. This is what will give us maximum sharing within the codebase.